### DISCUSSION TITLE
Segment for A/B testing

### DISCUSSION THREAD

**Discussion by looee1q** [rees46/android-sdk#91] [General]
Я не очень понимаю, как в SDK должна происходить работа с сегментом [segment](https://github.com/rees46/android-sdk/blob/fcc159117d985510f27e11af9a6cf92413d8158c/personalization-sdk/src/main/kotlin/com/personalization/SDK.kt#L43) для A/B тестирования. 

Сейчас при каждом входе в приложение генерируется новое значение сегмента [случайным образом](https://github.com/rees46/android-sdk/blob/fcc159117d985510f27e11af9a6cf92413d8158c/personalization-sdk/src/main/kotlin/com/personalization/sdk/data/repositories/preferences/PreferencesDataSource.kt#L77): либо `A`, либо `B`.

При этом значение сегмента не сохраняется в `sharedPreferences` (такого метода просто не существует в [PreferencesDataSource](https://github.com/rees46/android-sdk/blob/fcc159117d985510f27e11af9a6cf92413d8158c/personalization-sdk/src/main/kotlin/com/personalization/sdk/data/repositories/preferences/PreferencesDataSource.kt#L7C7-L7C28)). Однако, не смотря на это, мы все-равно стараемся [получить значение сегмента из SharedPreferences](https://github.com/rees46/android-sdk/blob/fcc159117d985510f27e11af9a6cf92413d8158c/personalization-sdk/src/main/kotlin/com/personalization/sdk/data/repositories/preferences/PreferencesDataSource.kt#L37). Но поскольку там всегда пусто, то присваивается значение по умолчанию - которое и есть случайное значение. Так ведь не должно быть? 

И как в общем должна обеспечиваться работа с сегментом для A/B тестирования в нашем SDK?

**Comment by looee1q** [comment]
@noff, можешь, пожалуйста, подсказать, как описанная выше функциональность должна быть реализована в SDK с точки зрения продукта? И как она в принципе задумывалась.

**Reply by noff** [reply]
Точного ответа не дам, но можно с @Nelfimov посмотреть логику сохранения сегмента в JS SDK. Если он устанавливается на короткое время, тогда в мобилке можно оставить текущую логику, только не пытаться получить сегмент из хранилища, а генерировать новый при каждом запуске.

Если же в JS SDK сегмент ставится на длительный срок, то в мобилке реализовать так же.
